### PR TITLE
Staging and idempotency

### PR DESCRIPTION
**Pull request by mbredif**
This PR generalizes the staging mode and idempotent behavior introduced in #5 for the orimatis entrypoint to the autocal and blinis entrypoints

### CODE REVIEW

File: micmac_li3ds/api.py [hunk 1]
```diff
@@ -7,21 +7,47 @@
...
+        if api_url and api_key:
+            self.api_url = api_url.rstrip('/')
+            self.headers = {
+                'Accept': 'application/json',
+                'X-API-KEY': api_key
+                }
+
+        else:
+            if api_key:
+                err = 'Error: no api url provided'
```
Comment (elemoine) [inline micmac_li3ds/api.py]: The error message is not consistent with the test.

What don't you just use `api_url` to determine whether staging is activated or not? Something like this:

```py
    if api_url:
        if not api_key:
             err = 'Error: no api key provided'
             raise ValueError(err)
        self.api_url = api_url.rstrip('/')
        self.headers = {
            'Accept': 'application/json',
            'X-API-KEY': api_key
        }
    else:
        self.staging = {
            …
        }
```

File: micmac_li3ds/api.py [hunk 2]
```diff
@@ -44,6 +74,11 @@ def get_object_by_id(self, typ, obj_id):
         raise RuntimeError(err)
 
     def get_object_by_name(self, typ, obj_name):
+        if self.staging is not None:
```
Comment (elemoine) [inline micmac_li3ds/api.py]: Why not just use `if self.staging:` here as in the other functions?

File: micmac_li3ds/api.py [hunk 3]
```diff
@@ -112,24 +156,22 @@ def get_or_create_object(self, typ, obj, keys, log):
...
                 if obj[key] != got[key]:
                     err = 'Error: "{}" mismatch in {} with id {:d} ' \
                           '("{}" vs "{}")' \
                           .format(key, typ, obj['id'], obj[key], got[key])
                     raise RuntimeError(err)
 
-            log.info('{} "{}" found in database with id {:d}.'
-                     .format(typ, got['name'], got['id']))
-            return got
+            return got, 'i'
```
Comment (elemoine) [inline micmac_li3ds/api.py]: Is the second element of the tuple (`i`, `=` or `+`) actually used somewhere?

File: micmac_li3ds/xmlutil.py [hunk 4]
```diff
@@ -65,4 +65,14 @@ def child_floats(parent, name):
...
-    return list(map(lambda n: child_float(parent, prefix+n+suffix), names))
+    return [child_float(parent, prefix+n+suffix) for n in names]
+
+
+def child_floats_split(parent, name):
+    node = child(parent, name)
+    try:
+        return [float(v) for v in node.text.split()]
+    except ValueError:
+        err = 'Error: {} tag ' \
```
Comment (elemoine) [inline micmac_li3ds/xmlutil.py]: Minor, you may want to place double-quotes around `{}` as in the other functions above.

**Comment by elemoine** [comment]
LGTM. Except that you did not answer my question regarding `i`, `=` and `+`. Is this actually used?

**Comment by mbredif** [comment]
It is used as an informational text shown at the beginning of the logging of import entrypoints. (In get_or_create)

**Pull request merged**